src/R/Components/Impl/PackageManager/Implementation/ViewModel/RPackageManagerViewModel.cs

```diff
@@ -184,17 +184,17 @@ internal class RPackageManagerViewModel : BindableBase, IRPackageManagerViewMode
 
         private async Task UninstallAsync(IRPackageViewModel package) {
             _coreShell.AssertIsOnMainThread();
-            if (_selectedTab == SelectedTab.InstalledPackages) {
-                IsLoading = true;
-            }
+            IsLoading = true;
```

Comment:
Switching to available quickly seems to work since I think, available requests state and since uninstall process already started, enumeration package state happens well after uninstallation started. Although yes, theoretically, depending on timing, state may not be correct. I think maybe model update should fire event on main thread when collections change so active tab can update itself. Parser does this guaranteeing that when tree change event fires it will be processed in UI thread i.e by the UI element that is currently active.